Develop

zxdcm opened the pull request.

zxdcm committed:
Navigations properties removed

zxdcm committed:
Domain services (Hotel & Booking) added.
CompleteBooking, CanBookExtraServices, HasAvailableRooms methods implemented

zxdcm committed:
Naming conventions fixes

zxdcm committed:
Unused method removed

zxdcm committed:
Queries dir structure reworked.
Base queries added.

zxdcm committed:
Errors resource file renamed

zxdcm committed:
EditHotelExtraServices command added.

zxdcm committed:
Commands now returns Ids as value.
Execute method now async (ExecuteAsync)

zxdcm committed:
Result now stores int Value;
NullEntityError now use string from resource file.
List of errors field added.

zxdcm committed:
Dispatcher became async.

zxdcm committed:
BookExtraService command added.
BookRoom command handler execute method splited.

zxdcm committed:
Execute method renamed to ExecuteAsync.
Dto renamed to RoomDto.

zxdcm committed:
Resource file renamed

zxdcm committed:
BookingConfiguration class added.

zxdcm committed:
Controller methods names are fixed.

zxdcm committed:
AddHotel & EditHotel methods now specifies resource location.
Commented parts of useless code are removed.
Unused imports removed.

zxdcm committed:
Dependencies registred

ssachkovskaya commented on BookingSystem/BookingSystem.Commands/Commands/HotelCommands/Commands/EditHotelCommand.cs:
```diff
@@ -28,17 +28,18 @@ public EditHotelCommandHandler(BookingWriteContext dataContext, IMapper mapper)
 
-            var hotel = await _dataContext.Hotels.FindAsync(dto.HotelId);
+            var hotel = await _dataContext.Hotels.FindAsync(hotelDto.HotelId);
             if (hotel == null)
```
Why do you use isActive for hotel, but isAvailable for extra services. Are there different delete operations?

zxdcm replied:
Fixed.

ssachkovskaya commented on BookingSystem/BookingSystem.Commands/Commands/HotelCommands/Commands/EditHotelExtraServicesCommand.cs:
```diff
@@ -0,0 +1,48 @@
+
+namespace BookingSystem.Commands.Commands.HotelCommands.Commands
+{
+    public class EditHotelExtraServicesCommand : ICommand<Result>
```
where are you going to use this command taking into account you already have add/edit/delete extra service commands?

ssachkovskaya commented on BookingSystem/BookingSystem.Commands/Commands/HotelCommands/DTOs/NewHotelDto.cs:
```diff
@@ -4,7 +4,7 @@ public class NewHotelDto
         public string Name { get; set; }
         public string Address { get; set; }
-        public int CountryId { get; set; }
+//        public int CountryId { get; set; }
```
remove?

ssachkovskaya commented on BookingSystem/BookingSystem.Commands/Infrastructure/Result.cs:
```diff
@@ -1,44 +1,56 @@
+        public static Result Ok()
+            => new Result(null, true, null);
+
+        public static Result Ok(int id) => new Result(id, true, null);
```
this should be written in 2 lines according your code style

ssachkovskaya commented on BookingSystem/BookingSystem.Queries/Queries/BookingQueries/Queries/BookingDetailsQuery.cs:
```diff
@@ -24,19 +21,28 @@ public BookingDetailsQuery(int bookingId)
+                {
+                    BookingId = b.BookingId,
+                    RoomId = b.RoomId,
+                    Status = b.Status.ToString(),
```
Are you sure you need to get the status as a string?

ssachkovskaya commented on BookingSystem/BookingSystem.Queries/Queries/CityQueries/Queries/ListCitiesQuery.cs:
```diff
@@ -42,7 +42,7 @@ public async Task<IEnumerable<CityView>> Execute(ListCitiesQuery query)
                     CityName = city.Name,
                     CountryId = city.CountryId,
-                    CountryName = city.Country.Name,
+                    //CountryName = city.Country.Name,
```
remove

ssachkovskaya commented on BookingSystem/BookingSystem.Queries/Queries/CityQueries/Queries/CityDetailsQuery.cs:
```diff
@@ -0,0 +1,45 @@
+            _dataContext = dataContext;
+        }
+
+        public async Task<CityView> Execute(CityDetailsQuery query)
```
ExecuteAsync?

ssachkovskaya commented on BookingSystem/BookingSystem.Queries/Queries/CountryQueries/Queries/CountryDetailsQuery.cs:
```diff
@@ -1,14 +1,11 @@
-namespace BookingSystem.Queries.CountryQueries
+namespace BookingSystem.Queries.Queries.CountryQueries.Queries
 {
     public class CountryDetailsQuery : IQuery<Task<CountryView>>
```
Can you please remind me what case will be when you need the single city/country details?

ssachkovskaya commented on BookingSystem/BookingSystem.Queries/Queries/HotelQueries/Queries/HotelDetailsQuery.cs:
```diff
@@ -0,0 +1,115 @@
+                    CountryName = country.Name,
+                    CityName = city.Name,
+                    IsActive = hotel.IsActive,
+//                    ExtraServices = service.Select(se => new ExtraServiceView()
```
Is it still in work? If no, please, remove

ssachkovskaya commented on BookingSystem/BookingSystem.ReadPersistence/ReadModels/Booking.cs:
```diff
@@ -1,26 +1,20 @@
-        }
+        public Booking() { }
 
         public int BookingId { get; set; }
```
Can investigate if properties can be with private set?

ssachkovskaya commented on BookingSystem/BookingSystem.WebApi/Controllers/BookingController.cs:
```diff
@@ -49,18 +46,23 @@ public async Task<IActionResult> GetBooking(int id)
-            return FromResult(result);
+            var result = await _commandDispatcher.DispatchAsync(new BookRoomCommand(booking));
+            if (result.IsSuccessful == false)
+                return BadRequest(result);
```
BadRequest is usually used for wrong input. Think about other types of errors

Commit:
IsAvailable field of ExtraService entity renamed to IsActive

Commit:
Solution cleared:
1. Unused imports removed.
2. Commented parts of code removed.
3. Code style fix (Result.cs Ok static method)

Commit:
BookingView no more return status as string.
BookingStatus now is a number.

Commit:
BadRequest result replaced by UnprocessableEntity.
Unused controller methods removed.

Commit:
Unused imports removed pt.2
(somewhy this changes were missed at prev commit)

Commit:
PasswordHasher added

Commit:
SignIn and SignUp commands added.

Commit:
Style fixes:
1. dto renamed to bookingDto
2. FirstOrDefault call replaced by Find

Commit:
QueryHandler Execute method refactored to ExecuteAsync

Commit:
HotelQuery partially fixed.

Commit:
ReadModels setter now private

Commit:
Missed filtering by name added.

Commit:
CountryId column removed from EditedHotelDto;
CityId validation added.

Commit:
typos in cityname & countryname fixed.

Commit:
ExtraServiceController added

Commit:
Conventions fixes. (async methods)

ssachkovskaya commented on BookingSystem/BookingSystem.Commands/Commands/AccountCommands/Commands/SignInCommand.cs:
```diff
@@ -0,0 +1,52 @@
+
+            if (_hasher.VerifyPassword(user.PasswordHash, userDto.Password))
+            {
+                return Result.Ok(user.UserId);
```
I would make Ok result the last one in this method in order not to interrupt the chain of error results

ssachkovskaya commented on BookingSystem/BookingSystem.Commands/Commands/BookingCommands/Commands/BookExtraServicesCommand.cs:
```diff
@@ -0,0 +1,55 @@
+                return Result.NullEntityError(nameof(Booking), bookingDto.BookingId);
+
+            var result = await _bookingService.CanBookExtraServicesAsync(booking, bookingDto.ExtraServicesIds);
+            if (!result)
```
Result can be any type, it is not clear that this result is boolean. Please make it clear with the variable name

ssachkovskaya commented on BookingSystem/BookingSystem.Queries/Queries/HotelQueries/Queries/HotelDetailsQuery.cs:
```diff
@@ -0,0 +1,149 @@
+                into gr
+                select gr;
+            var hotelGr = await hotelViews.FirstOrDefaultAsync();
+            return new HotelDetailsView
```
isn't it possible to select HotelDetailsView as a part of query and then apply FirstOrDefaultAsync?

ssachkovskaya commented on BookingSystem/BookingSystem.Commands/Commands/AccountCommands/Commands/SignUpCommand.cs:
```diff
@@ -0,0 +1,59 @@
+            if (userExist)
+                return Result.Error(string.Format(ErrorsResources.UserAlreadyExist, userDto.Email));
+
+            var user = _mapper.Map<User>(userDto);
```
Are you planning to have any validation for empty email, too short password and etc.?

zxdcm committed:
Style fixes

zxdcm committed:
result variable name changed to canBook

zxdcm committed:
Ok result made last one.

zxdcm committed:
Unused command removed.
